### ISSUE TITLE
Is is required to specify "assay" in the "standard workflow" functions?

### ISSUE THREAD

**Issue by CodeInTheSkies** [satijalab/seurat#2163]
Hi there,

I have a basic question about the "assay" organization and definition in the Seurat v3 object.  

I see that when we first create the Seurat v3 object, the "RNA" assay gets created and the data, by default, falls into the RNA slot.  The `active.assay` becomes "RNA" at this point.  Then, when we use `SCTransform` downstream, I see that `active.assay` changes to "SCT".  My question is, after this step, when we use `FindNeighbors` and `FindClusters`, does the default assay get recognized as SCT, since `active.assay`=SCT?  

In other words, although these functions have an "assay" parameter to be specified if desired, they will look at `active.assay` and take whatever assay is specified there as their input (if assay is not specified, that is).  Is this understanding correct?

Also, is there a function then, to specify the `active.assay`, similar to the way we set "idents" to any `meta.data` column we want?

Thank you!

**Comment by CodeInTheSkies** [comment]
Just adding an additional question/"request for confirmation" on this thread:

I'm trying to understand what exactly is the need for an "assay" parameter in the functions `FindNeighbors` and `FindClusters`, given that the "pca" slot is actually **_outside_** the assay slots.  Initially, it gave me the impression that setting the assay is irrelevant as the pca slot is outside the assay slot.  But then, I now notice that within the pca slot, there is a mention of which assay the pca was computed from.  So, it looks like, we can have multiple pca sets within the pca slot simultaneously, each set coming from different assays.  So, if we have such a scenario, then we can specify which assay's pca values should be used during a particular `FindNeighbors` function call.  Is this understanding correct?

Thanks a lot!

**Comment by andrewwbutler** [comment]
Hi,

Yes, that is correct. By default, Seurat will use which ever assay is currently set as the default or "active.assay". Many functions also include an `assay` parameter which can be set to override this default behavior as you say. 

The function to switch the `active.assay` is called `DefaultAssay`. For example to switch the default assay back to RNA:

```R
DefaultAssay(object) <- "RNA"
```

For `FindNeighbors`, we support either computing the neighbor graph in a reduced dimension space (e.g. PCA space) or directly on the underlying data itself. Therefore, the `assay` parameter is only used in `FindNeighbors` if you are computing on the data itself in order to know which assay to pull that expression data from. If you supply the `reduction` parameter, then it will build the graph based on the dimensionally reduced space (and the `assay` parameter doesn't affect things here). 

There is no "pca" slot. There is however a slot that stores the results of running dimensional reductions (such as PCA), so you could store for example a PCA computed on the "RNA" assay and one computed on the "SCT" assay as separate items in that `reductions` slot. These need to be named differently though and you would distinguish between them in the `FindNeighbors` call by what you provided to that function's `reduction` parameter.
